### PR TITLE
Introduction to Application Performance Management

### LINKED ISSUE #2537: Introduction to Application Performance Management (APM) and its significance
### Proposed title of article
Introduction to Application Performance Management (APM) and its significance

### Introduction paragraph (2-3 paragraphs):
The performance of computer application systems directly affects the output of businesses. For instance, a client getting irritated by slow application systems. Such clients are easy to lose when a company's service application system performs poorly. To prevent such financial and reputational losses, APM is used to ensure IT applications run smoothly and effectively. It is also used to maximize the utility of applications by, among other functions, detecting application issues that may cause poor performance or crushing of applications before it happens.
 
This article will enlighten application users with all there is to know about APM and its significance in financial production. It will focus on APM as a tool to measure performance and outline how the response time of applications is affected by peak load and how the applications use computer resources to get tasks completed. It describes the dimensions of APM in detail and outlines the challenges that may be experienced during its application.

### Key takeaways: 

The key take-homes of this article include but are not limited to;
•	Understanding of what Application Performance Management (APM) is and its purpose in IT applications
•	Knowledge of how performance metrics are used to measure the performance of applications
•	The ability to apply Application Performance Management (APM) in the identification of causes of delay in computer applications
•	Understanding of challenges that may be experienced in the implementation of Application Performance Management (APM), and knowledge of how to overcome them.

### References: 
n/a

### Templates to use as guides
n/a

### PR DESCRIPTION
**Pull request by Ernest715**
this fixes #2537

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by Ernest715** [comment]
Hi @collins-a, any amendments you recommend for this article from my end?

**Comment by collins-a** [comment]
> Hi @collins-a, any amendments you recommend for this article from my end? 

Hey @Ernest715, thanks for following up, I'll be sure to do a complete review over the weekend.

**Comment by Ernest715** [comment]
I will truely appreciate @collins-a

File: content/articles/introduction-to - application-performance-management/index.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+-   Application Discovery, Tracing, and Diagnostics (ADTD)
+
+-   Artificial Intelligence for IT Operations (AIOps)
+
+Digital Experience Monitoring (DEM) focuses on the relationship between users and applications. The data it collects and analyses are based on how users get affected by the application in use. It detects and helps solve problems that affect user experience directly. Examples of such problems include long application response time and user unfriendliness.
+
+On the other hand, Application Discovery, Tracing, and Diagnostics (ADTD) address the technical logistics of an application. This includes the runtime architecture. It manages applications by monitoring both hardware and software resources that form the digital environment to execute an application.
+
+Artificial Intelligence for IT Operations (AIOps) applies to platforms that analyze and learn machine language to enhance the performance of applications. It improves performance continuously by integrating and deploying core IT functions to suit different and more sophisticated tasks.
```
Comment (collins-a) [inline content/articles/introduction-to - application-performance-management/index.md:45]: Interesting stuff here. Consider delving a bit deeper into these dimensions of APM.

**Review by collins-a — COMMENTED** [review]
Great article introducing APM @Ernest715. Very polished as well. Just add a bit more content to the dimensions of APM and we're good to go. Ping me when ready.

Also, take note of how I have represented the links to the references, for future articles.

**Comment by hectorkambow** [comment]
plagiarism check done

File: content/articles/introduction-to - application-performance-management/index.md (hunk 1 above)
Comment (Ernest715) [reply content/articles/introduction-to - application-performance-management/index.md:45]: this is well noted @collins-a . i will work on it and get back to you shortly.
Thanks for the positive feedback.

**Comment by Ernest715** [comment]
@collins-a I have expounded on the dimensions of APM. kindly have a look at it and revert.
Thanks

File: content/articles/introduction-to - application-performance-management/index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+WordPress businesses prefer [New Relic](https://newrelic.com/ ), which charges $99 each month as of June 2021, while [ITRS](https://en.wikipedia.org/wiki/Semaphore_(programming)) is the tool of management for most banking entities. [Pinpoint](https://developers.redhat.com/blog/2018/10/22/introduction-to-linux-interfaces-for-virtual-networking#:~:text=A%20Linux%20bridge%20behaves%20like,VLAN%20filter%2C%20and%20multicast%20snooping.), [AppDynamics](https://searchnetworking.techtarget.com/definition/routing-table) and [Dynatrace](https://www.dynatrace.com/ ) are powerful monitoring tools and are used by large enterprises.
```
Comment (collins-a) [inline content/articles/introduction-to - application-performance-management/index.md:76]: Noticed that some of the links don't lead to the corresponding tools in this section.

**Review by collins-a — COMMENTED** [review]
Hey @Ernest715, thanks for making changes. Fix some of the APM tool links and we'll be good to go.

File: content/articles/introduction-to - application-performance-management/index.md (hunk 2 above)
Comment (Ernest715) [reply content/articles/introduction-to - application-performance-management/index.md]: an oversight on my part. This is now resolved @collins-a

**Review by collins-a — APPROVED** [review]
@Ernest715 I'm satisfied with the article. Keep this level of quality up. Perhaps you could consider looking at _application monitoring and observability_ in detail as a future article provided it hasn't been written already.

**Comment by Rmwamboi** [comment]
Sure,  I will look into that.

**Review by louisefindlay23 — APPROVED** [review]
Great article, @Ernest715. Final review complete. Make sure to have a look at my edits to inform future articles. 

**Note for future articles:** Please remove any spaces in the article folder and URLs. Also, please remove any parts at the end of URLs that are not needed such as #:~text=.

Thanks, @collins-a for peer reviewing.

**Pull request merged**